numbers/minMax.go

```diff
@@ -0,0 +1,101 @@
+package numbers
+
+func Max(a int, b int) int {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MaxInt32(a int32, b int32) int32 {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MaxUint32(a uint32, b uint32) uint32 {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MaxInt64(a int64, b int64) int64 {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MaxFloat64(a float64, b float64) float64 {
+	if a >= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func Min(a int, b int) int {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MinInt32(a int32, b int32) int32 {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MinUint32(a uint32, b uint32) uint32 {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MinInt64(a int64, b int64) int64 {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
+func MinFloat64(a float64, b float64) float64 {
+	if a <= b {
+		return a
+	} else {
+		return b
+	}
+}
+
```

Comment:
Are these triple functions commonly used? Would it be better to just have a function for a slice of ints? Not sure what is best stylistically moving forward

Reply:
I am against using a slice for this function. You would have to start looping through the slice and you have to allocate additional memory to keep track of the the Min/Max. Using individual numbers also ensures you allocate memory on the stack (at least that is my numberstanding).

Reply:
No, i am against using slices for this. You also have to program a loop and also have to allocate additional memory to keep track of the current max/min, whereas here no additonal memory is needed to return an answer. When you work with simple data types (like ints and strings) memory allocation is more likely on the stack. When you start to use more slices, maps or complex structs, if you are sloppy or not careful, the memory escapes to the heap (at least that is my understanding).

Reply:
I think it makes sense to keep these functions for the reasons Eric gave.  However, in the future we could add versions of min/max that operate on a slice if that would be useful for people.

Reply:
The slice version of min/max essentially would look the same as a sort, either ascending or descending order, returning either the head or tail, is that right?

Reply:
Not necessarily. A min/max would require far fewer comparisons than a true sort.